src/index.js

```diff
@@ -59,14 +74,23 @@ function _checkMasterHelper(timeout=100) {
         resolve();
       })
       .catch((err, resp) => {
-        log.warnThrottle(60000, `Unable to register with master node [${rosNode.getRosMasterUri()}]: master may not be running yet. Will keep trying.`);
-        localHelper(resolve);
+        let error = `Unable to register with master node [${rosNode.getRosMasterUri()}]: master may not be running yet.`
+        if (Time.now().secs - startTime >= maxTimeout && !(maxTimeout < 0) ){
+          error += ' Timed out! Stopping...';
+          log.error(error);
```

Comment:
can you log a separate message here rather than manipulating the 'base' error?
`Timed out before registering with master node [${rosNode.getRosMasterUri()}]: master may not be running yet.`